lib/run.sh

```diff
@@ -1,6 +1,14 @@
 #!/bin/bash
 
+source lib/print.sh
+
 run() {
   echo -e "\x1B[1;34m==\$ $*\x1B[0m"
   eval "$*"
 }
+
+run_with_retry() {
+  if ! run "$*" && ask_Yn "Something went wrong. Try again?"; then
```

Comment:
It might be unclear whether answering yes to "Try again?" means restart the entire install script or retry the last step - the latter, which it is, sounds less scary! And likewise whether answering no means exit the script (and potentially miss something important) or continue to the next step

```suggestion
  if ! run "$*" && ask_Yn "Something went wrong. Try the last step again?"; then
```

Reply:
This is a good shout. Thank you.